app/javascript/admin/controllers/data_update_script_controller.js

```diff
@@ -0,0 +1,114 @@
+import { Controller } from 'stimulus';
+
+export default class DataUpdateScriptController extends Controller {
+
+  forceRun(event) {
+    event.preventDefault();
+    const id         = event.target.dataset.value;
+    let statusColumn = document.getElementById(`data_update_script_${id}_status`);
+    let runAtColumn  = document.getElementById(`data_update_script_${id}_run_at`);
+
+    this.displayLoadingIndicators(statusColumn, runAtColumn);
+    this.forceRunScript(id, statusColumn, runAtColumn);
+  }
+
+  displayLoadingIndicators(statusColumn, runAtColumn) {
+    runAtColumn.innerHTML  = 'loading..';
+    statusColumn.innerHTML = '';
+  }
+
+  forceRunScript(id, statusColumn, runAtColumn) {
+    fetch(`/admin/data_update_scripts/${id}/force_run`, {
+      method: 'POST',
+      headers: {
+       'X-CSRF-Token': document.querySelector("meta[name='csrf-token']").content,
+      },
+      credentials: 'same-origin'
+    })
+    .then(response => {
+      if(response.ok) {
+        this.pollForScriptResponse(id, statusColumn, runAtColumn);
+      } else {
+        response.json().then((response) => {
+          this.setErrorBanner(
+            runAtColumn,
+            statusColumn,
+            `Data Update Script ${id} - ${response.error}`
+          );
+        });
+      }
+    })
+  }
+
+  pollForScriptResponse(id, statusColumn, runAtColumn) {
+    let counter = 0;
+    let pollForStatus = setInterval(() => {
+      counter++;
+      this.checkForUpdatedDataScript(id, runAtColumn, statusColumn).then((updatedDataScript) => {
+        if (updatedDataScript) {
+          if(updatedDataScript.status) {
+            // when we've stopped polling because we've received a status
+            // and not because we've received an error.
+            runAtColumn.innerHTML = updatedDataScript.run_at;
+            statusColumn.innerHTML = `${updatedDataScript.status}`;
+            if(updatedDataScript.error) {
+              statusColumn.innerHTML += `<div class='fs-xs'> ${updatedDataScript.error}</div>`
+            }
+            if(updatedDataScript.status === "succeeded") {
+              document.getElementById(`data_update_script_${id}_row`).classList.remove("alert-danger");
+              if(document.getElementById(`data_update_script_${id}_button`)) {
+                document.getElementById(`data_update_script_${id}_button`).remove();
+              }
+            }
+          }
+          clearInterval(pollForStatus);
+        }
+      });
+      if ( counter > 20 ) {
```

Comment:
Not likely but the downside with this mechanism is that it will have 1 timer for each retried script, instead of checking them as a batch. Just an idea for future improvements

Reply:
Thanks Rhymes, this makes sense too, and I've noted it for a future improvement.